pkg/blobinfocache/internal/prioritize/prioritize.go

```diff
@@ -37,6 +37,16 @@ func (css *candidateSortState) Less(i, j int) bool {
 	xi := css.cs[i]
 	xj := css.cs[j]
 
+	// If i has location unknown and j has known location
+	// always treat i as lesser than j or vice-versa so the
+	// candidate with location unknown alaways gets lower priority.
```

Comment:
(This comment could be a bit higher-level, “unknown-location entries are always the very last”, or something like that. But see elsewhere.)

Reply:
This part is removed.